### ISSUE TITLE
[HOLD #30668] [$500]  Chat - Unpinned deleted tasks assigned by another user are shown above other pinned messages

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#30798]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.95-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Login with Account A on new Expensify and assign a task to Account B
2. Cancel the task
3. Login with Account B > find the Deleted task in the LHN
Tester login credential: [gatantm+82@gmail.com](mailto:gatantm+82@gmail.com,)    [gatantm+112@gmail.com](mailto:gatantm+112@gmail.com)
4. If you don't see it above the pinned messages, click on the [Deleted task] message again and eventually it displays above the pinned chats

## Expected Result:
Unpinned messages appear under pinned chats on LHN

## Actual Result:
In the new build the Assigned tasks from another account that were canceled appear above or between pinned messages when they were never pinned

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/51066321/5bdb30fa-f53e-4303-af2d-6145801990a1

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f5bd596fa8bc9e1a</li>
        <li>Upwork Job ID: 1721590933982339072</li>
        <li>Last Price Increase: 2023-11-06</li>
<li>Automatic offers: </li>
<ul>
<li>s77rt | Reviewer | 27545058</li>
</ul></li>
    </ul>
</details>

**Comment by Christinadobrzyn** [comment]
I can't reproduce this...  the deleted task isn't showing as pinned in the LHN

- Login with Account A on new Expensify and assign a task to Account B
- Cancel the task
- Login with Account B and check where the task is on LHN


https://github.com/Expensify/App/assets/51066321/21d6d7f0-179f-41e3-9595-3189fd774279

**Comment by kbecciv** [comment]
@Christinadobrzyn Can you create a task and assign it to user B using the 'share somewhere' option?

**Comment by Christinadobrzyn** [comment]
I can reproduce with the steps in the OP. @kbecciv Should I also test the share somewhere option for this?

**Comment by Christinadobrzyn** [comment]
This might be related to https://github.com/Expensify/App/pull/28431 - I'll ask

**Comment by AndrewGable** [comment]
I don't think that PR is merged or deployed yet 🤔

**Comment by AndrewGable** [comment]
I cannot reproduce 🤔 

For me the deleted task never shows up in the LHN.

Example (on staging web):
1. andrew@expensify.com sends a task to andrew+empty@expensifail.com
2. andrew+empty@expensifail.com gets the task
3. andrew+empty@expensifail.com has a pinned chat
4. andrew@expensify.com cancels task
5. andrew+empty@expensifail.com no longer sees task in LHN or chat

**Comment by AndrewGable** [comment]
When clicking on the deleted task/thread, I see it ordered correctly:

![Screenshot 2023-11-03 at 11 05 18 AM](https://github.com/Expensify/App/assets/2838819/18a49065-508c-43fa-a5a4-ee07058fd908)

**Comment by AndrewGable** [comment]
Asking for a retest: https://expensify.slack.com/archives/C9YU7BX5M/p1699031208685439

**Comment by kbecciv** [comment]
The issue is consistently reproducible.
Providing the reducible steps:
1. Sign in with Account A
2. Create Task and assign User B
3. Cancel the task from Account A
4. Log in with Account B and check LHN
Note:  User B must open the task to make it read

https://github.com/Expensify/App/assets/93399543/beb3757a-fd13-43a2-8049-5c4aa2c9c084

**Comment by AndrewGable** [comment]
Talking with @puneetlath to see if this PR is related https://github.com/Expensify/App/pull/30655

**Comment by puneetlath** [comment]
@waterim would you be able to look into this one? I think it's related to our GBR changes.

**Comment by AndrewGable** [comment]
I pushed a revert branch named `andrew-revert-gbr` and @puneetlath thinks it fixed the issue, so I do agree it seems related to the PR linked above. @waterim can you let us know what you find out?

I discussed with @puneetlath and @JmillsExpensify and we decided not to block the deploy on this as Tasks is a relatively small feature in general.

**Comment by waterim** [comment]
Hello,Im Artem and I can take this one
(Already started the investigation)

**Comment by SrR0b0** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Canceled task message appears above pinned tasks in the assignee LHN.

### What is the root cause of that problem?

During the ordering of the reports to be displayed in the LHN [here](https://github.com/Expensify/App/blob/d1bc5e9104642387cad6e6aa874edd05ad8967ff/src/libs/SidebarUtils.ts#L113) the `requiresAttentionFromCurrentUser` function returns `true` when it should return `false` and it causes the canceled task report to be added to the `pinnedAndGBRReports` array which contains the arrays which will be displayed first and since this array is later alphabetically sorted the task usually ends up at the top.

The `requiresAttentionFromCurrentUser` function returns `true` because the `isWaitingForAssigneeToCompleteTask` function unexpectedly returns `true` which in turn does it because the `isOpenTaksReport` function also does it and, finally, it happens because the `isCanceledTaskReport` function returns `false` when it should, obviously, return `true`.

`isCanceledTaskReport` returns false because it checks for the report's property `isDeletedParentAction` which should be true but is `undefined` during execution even though the property is properly setted to `true` in Onyx db.

And here is the actual root cause. The property `isDeletedParentAction` is `undefined`  for all reports passed to `getOrderedReportIDs` by the `SidebarLinksData` because this component uses a selector called `chatReportSelector` which "narrow down the data from Onyx to just the properties that we want to trigger a re-render of the component" as stated in a comment and it happens that the `isDeletedParentAction` property was deemed unecessary and removed by the selector.

### What changes do you think we should make in order to solve the problem?
Simply updating the `chatReportSelector` selector function to keep the `isDeletedParentAction` property solves the issue. 

### What alternative solutions did you explore? (Optional)

**Comment by puneetlath** [comment]
Sorry @SrR0b0 this issue no longer available. @waterim is going to work on it since it relates to changes in a previous PR of his.

**Comment by waterim** [comment]
@puneetlath From my research, we have the same behaviour on new.expensify version, deleted task appears above pinned messages and with constant green dot

**Comment by waterim** [comment]
Okay, I found the main issue.
This is because of our change in moving isWaitingForAssigneeToCompleteTask to the requiresAttentionFromCurrentUser function. In accepts parentReportAction and it needs to have deleted-task action, but it doesn't have any parentReportAction when we are using requiresAttentionFromCurrentUser from getOrderedReportIDs in SidebarUtils, because we dont have any parent actions in there, will think about the best solution here

**Comment by puneetlath** [comment]
Ah interesting. How's the investigation going?

**Comment by s77rt** [comment]
Any updates here @waterim?

**Comment by waterim** [comment]
@s77rt Working on it!
Callstack had a day off on Friday

**Comment by waterim** [comment]
@puneetlath I think this issue will be fixed after this [issue](https://github.com/Expensify/App/issues/30668)
Because now the issue is because when we are calling `requiresAttentionFromCurrentUser` on OptionRowLHN.js it has `parentReportAction`.
But when we are calling it from SidebarUtils.ts file we don't have any specific `parentReportAction` available to pass and thats why `isWaitingForAssigneeToCompleteTask` will be always true.
But with refactor, depending on type I can prevent the call of isWaitingForAssigneeToCompleteTask for reports, not options.

What do you think?

**Comment by puneetlath** [comment]
Ok sounds good. I'll go ahead and put this on hold for that issue.

**Comment by Christinadobrzyn** [comment]
Hi @puneetlath it looks like https://github.com/Expensify/App/issues/30668 might be in a pre-PR stage. Do you think it'd be okay to move this GH to `weekly` while it's on hold?

**Comment by Christinadobrzyn** [comment]
I'm going ooo until Monday Nov 27th so going to assign another BZ teammate to monitor this until I'm back. Thanks!

Status: Hold for https://github.com/Expensify/App/issues/30668

**Comment by Christinadobrzyn** [comment]
Going to test this again tomorrow but I'm pretty certain it's still happening.

**Comment by Christinadobrzyn** [comment]
yes, I can still reproduce this based on the steps in the OP. It looks like it's above the pinned Concierge chat in the LHN. 

<img width="1200" alt="image" src="https://github.com/Expensify/App/assets/51066321/c35fec03-95b9-4d4c-9bca-a1b75e1d35f7">

**Comment by Christinadobrzyn** [comment]
So I think we're back to accepting proposals for this. Let me know if that's not right @s77rt

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - Unpinned deleted tasks assigned by another user are shown above other pinned messages

### What is the root cause of that problem?
The deleted task report still be GBRReport

### What changes do you think we should make in order to solve the problem?
We need to pass parentReportAction to requiresAttentionFromCurrentUser function

### What alternative solutions did you explore? (Optional)

**Comment by s77rt** [comment]
@Christinadobrzyn This is still hold on https://github.com/Expensify/App/issues/30668. Not accepting further proposals since @waterim is already assigned to take care of this (in case the other linked issue didn't fix it).

cc @DylanDylann

**Comment by s77rt** [comment]
Still on hold

**Comment by Christinadobrzyn** [comment]
I think I'm going to move this to weekly as we're still waiting on this [PR](https://github.com/Expensify/App/pull/32140) that hasn't been deployed yet. @s77rt let me know if you'd prefer this is still `daily`

**Comment by s77rt** [comment]
Weekly sounds good to me as well 👍

**Comment by Christinadobrzyn** [comment]
monitoring https://github.com/Expensify/App/pull/32140

**Comment by DylanDylann** [comment]
@Christinadobrzyn @s77rt [This PR](https://github.com/Expensify/App/pull/32140) is merged. Let's remove hold

**Comment by DylanDylann** [comment]
After re-check on the latest main, I see that after canceling the task the GRB still displays 

https://github.com/Expensify/App/assets/141406735/f0729f1b-6153-4cc0-b68e-d1b5fe0b6cc4

From my viewpoint, I think the GRB should disappear immediately after canceling the task

So that I post this proposal to fix this bug 


## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After canceling the task the GRB still displays on LHN

### What is the root cause of that problem?
After canceling the task the GRB still displays on LHN, The pusher return taskReport.isDeletedParentAction === true . But we don't add this field in this function
https://github.com/Expensify/App/blob/7f7cfb318bdb073cbaf478843255c2d5f9423d12/src/libs/SidebarUtils.ts#L232

### What changes do you think we should make in order to solve the problem?
in this function
https://github.com/Expensify/App/blob/7f7cfb318bdb073cbaf478843255c2d5f9423d12/src/libs/SidebarUtils.ts#L232

we should return  `isDeletedParentAction`
```
    result.isDeletedParentAction = report.isDeletedParentAction;

```

We also need to do the same thing in the chat report selector here
https://github.com/Expensify/App/blob/7f7cfb318bdb073cbaf478843255c2d5f9423d12/src/pages/home/sidebar/SidebarLinksData.js#L170
should add
```
        isDeletedParentAction: report.isDeletedParentAction

```

### What alternative solutions did you explore? (Optional)
NA

### Result

https://github.com/Expensify/App/assets/141406735/ea023fd8-2a72-46ca-a9e8-8afa750bd81d

**Comment by s77rt** [comment]
@DylanDylann This issue does not have `Help Wanted` label. @waterim is already assigned to work on this issue

**Comment by s77rt** [comment]
@Christinadobrzyn Please lift the hold since https://github.com/Expensify/App/pull/32140 is merged now

**Comment by s77rt** [comment]
@waterim Please check this issue once you get a chance

**Comment by waterim** [comment]
@s77rt I can see that the issue here is "Unpinned deleted tasks assigned by another user are shown above other pinned messages", but this one was fixed and as I understand we want to fix GBR for deleted task as well here, right?

**Comment by DylanDylann** [comment]
@s77rt @Christinadobrzyn Should we create a new issue for the bug that is mentioned [here](https://github.com/Expensify/App/issues/30798#issuecomment-1853461316) ? I think It also be a critical bug

**Comment by s77rt** [comment]
Since this is another bug I think it would be fair to create another issue for it.

**Comment by s77rt** [comment]
For this one I think we can just close it.

**Comment by DylanDylann** [comment]
@Christinadobrzyn @s77rt COuld you help to create a new issue for [this one](https://github.com/Expensify/App/issues/30798#issuecomment-1853461316)

**Comment by Christinadobrzyn** [comment]
So sorry for the delay @DylanDylann I created the new issue here- https://github.com/Expensify/App/issues/33477

should we close this one or work on them together?

**Comment by s77rt** [comment]
@Christinadobrzyn This can be closed
